pymongo/asynchronous/client_session.py

```diff
@@ -598,7 +601,10 @@ def _inherit_option(self, name: str, val: _T) -> _T:
 
     async def with_transaction(
         self,
-        callback: Callable[[AsyncClientSession], _T],
+        callback: Union[
+            Callable[[AsyncClientSession], Coroutine[Any, Any, _T]],
+            Callable[[AsyncClientSession], _T],
```

Comment:
Why are there two callables here?

Reply:
An artifact of the reasoning below, removed.